### PR TITLE
[IN-PROGRESS] GORA-698 Geode DataStore

### PR DESCRIPTION
**Pull request by himanshuacharya95**
# Pending Items Test Cases

### CODE REVIEW

File: gora-geode/src/main/java/org/apache/gora/geode/store/GeodeStore.java [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+import static org.apache.gora.geode.store.GeodeStoreParameters.*;
+
+
+public class GeodeStore<K, T extends PersistentBase> extends DataStoreBase<K, T> {
+
+    private ClientCache clientCache;
+    private Region<K, T> region;
+    private Properties geodeProperties;
+
+
```
Comment (kamaci) [inline gora-geode/src/main/java/org/apache/gora/geode/store/GeodeStore.java:30]: Please remove extra spaces

File: gora-geode/src/main/java/org/apache/gora/geode/store/GeodeStore.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        super.initialize(keyClass, persistentClass, properties);
+        String geodeHostName = (String) properties.get(GEODE_SERVER_HOST);
+        int portNumber = Integer.parseInt((String) properties.get(GEODE_SERVER_PORT));
+        clientCache = new ClientCacheFactory().addPoolLocator(geodeHostName, portNumber).create();
+        String userName = properties.getProperty(GEODE_USERNAME);
+        String password = properties.getProperty(GEODE_PASSWORD);
+        geodeProperties = properties;
+
+        Properties clientProperties = clientCache.getDistributedSystem().getProperties();
+        if (userName != null) {
```
Comment (kamaci) [inline gora-geode/src/main/java/org/apache/gora/geode/store/GeodeStore.java:44]: What if username is null and password is not? How about throwing an exception in that case?

File: gora-geode/src/main/java/org/apache/gora/geode/store/GeodeStore.java [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+            LOG.info("Geode datastore deleted {} rows from Persistent datastore.", deletedRows);
+            return deletedRows;
+        } catch (Exception e) {
+            throw new GoraException(e);
+        }
+    }
+
+    @Override
+    public Result<K, T> execute(Query<K, T> query) {
+
```
Comment (kamaci) [inline gora-geode/src/main/java/org/apache/gora/geode/store/GeodeStore.java:139]: remove extra space

File: gora-geode/src/main/java/org/apache/gora/geode/store/GeodeStore.java (hunk 1 above)
Comment (himanshuacharya95) [reply gora-geode/src/main/java/org/apache/gora/geode/store/GeodeStore.java]: Extra spaces removed

File: gora-geode/src/main/java/org/apache/gora/geode/store/GeodeStore.java (hunk 2 above)
Comment (himanshuacharya95) [reply gora-geode/src/main/java/org/apache/gora/geode/store/GeodeStore.java:42]: Added some code to throw exception

File: gora-geode/src/main/java/org/apache/gora/geode/store/GeodeStore.java (hunk 3 above)
Comment (himanshuacharya95) [reply gora-geode/src/main/java/org/apache/gora/geode/store/GeodeStore.java]: Extra space removed

File: gora-geode/src/main/java/org/apache/gora/geode/store/GeodeStore.java [hunk 4]
```diff
@@ -0,0 +1,187 @@
+package org.apache.gora.geode.store;
+
+import org.apache.geode.cache.*;
```
Comment (kamaci) [inline gora-geode/src/main/java/org/apache/gora/geode/store/GeodeStore.java:3]: Please not use wildcard imports

File: gora-geode/src/test/java/store/TestGeodeStore.java [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+import java.nio.charset.Charset;
+import java.time.Duration;
+
+import static org.junit.Assert.*;
+
+/**
+ * Tests extending {@link DataStoreTestBase}
+ * which run the base JUnit test suite for Gora.
+ */
+public class TestGeodeStore extends DataStoreTestBase {
```
Comment (kamaci) [inline gora-geode/src/test/java/store/TestGeodeStore.java:48]: Could you add more tests?

File: gora-geode/src/main/java/org/apache/gora/geode/store/GeodeStore.java (hunk 4 above)
Comment (himanshuacharya95) [reply gora-geode/src/main/java/org/apache/gora/geode/store/GeodeStore.java]: Replaced wildcard import statements with only required ones.

**Comment by djkevincr** [comment]
@CalvinKirs Do you have any input on the PR related to Geode work?

**Comment by CalvinKirs** [comment]
> @CalvinKirs Do you have any input on the PR related to Geode work?

I will take a look later :)

**Comment by djkevincr** [comment]
@himanshuacharya95 Can you please address the remaining request changes?

**Comment by djkevincr** [comment]
Theres some in complete work left in this PR, I will merge this PR for now and take things forward.

**Pull request merged**